Microsoft.Toolkit/Helpers/ObjectStorage/IFileStorageHelper.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Toolkit.Helpers
+{
+    /// <summary>
+    /// Service interface used to store data in files and folders.
+    /// </summary>
+    public interface IFileStorageHelper
+    {
+        /// <summary>
+        /// Determines whether a file already exists.
+        /// </summary>
+        /// <param name="filePath">Key of the file (that contains object).</param>
+        /// <returns>True if a value exists.</returns>
+        Task<bool> FileExistsAsync(string filePath);
+
+        /// <summary>
+        /// Retrieves an object from a file.
+        /// </summary>
+        /// <typeparam name="T">Type of object retrieved.</typeparam>
+        /// <param name="filePath">Path to the file that contains the object.</param>
+        /// <param name="default">Default value of the object.</param>
+        /// <returns>Waiting task until completion with the object in the file.</returns>
+        Task<T> ReadFileAsync<T>(string filePath, T? @default = default);
+
+        /// <summary>
+        /// Retrieves all file listings for a folder.
+        /// </summary>
+        /// <param name="folderPath">The path to the target folder.</param>
+        /// <returns>A list of file names in the target folder.</returns>
+        Task<IList<string>> ReadFolderAsync(string folderPath);
```

Comment:
Is this just in the specified directory or is it recursive (like grabbing ones in sub-folders too)? Is that an option? We should specify the default behavior at least.

Does it only retrieve files or does it return folder names in that list too?

Reply:
This one I had trouble with. On one side, I wanted to return a list of folders and files, but I couldn't come up with a concise way to differentiate files from folders without writing a bunch of file/folder item structures for this one purpose. So I decided on a simple list of strings, representing the files.

But now that we are discussing it, perhaps a `FileItem` struct is the right thing to do after all, with an `Id`, `Name`, and `Type` (File/Folder). Plus a `FileItemType` enum with `File` and `Folder` values. That would make it much easier to understand the response from ReadFolderAsync, and give us a more substantial object to return from SaveFile/FolderAsync.

Reply:
Yeah, I guess in some cases id == name? Like in OneDrive id could be meaningful, but not sure what that'd be for UWP. `handle` could be interesting if we go that route too if the file system provides a link to the file object itself that could be abstracted as just a genric implementation specific `object`, but I think that's probably getting too specific. I like the simplicity.

The other option is we just have the Folder\File enum type and return a list of `Tuple<FileItemType, string>`.

We should think about whether there's a recursive boolean parameter for iterating over the directories, otherwise the dev is going to have to write that all themselves scanning the return for folders and recalling the method each time. Would be handy. Think it's probably `false` by default though for simplicity?

Reply:
Actually, I think your Tuple idea is perfect. Even better than a struct 👍

I noticed that ApplicationData does have a recursive query option for files, but not folders. I added that functionality to the `ApplicationDataStorageHelper.FileExistsAsync` method, default is `false`.  Although, I don't think I can easily get the recursion working for folders without writing a bunch code to do repeat requests to traverse the directory. I think that is something the implementer can handle if they have the ability to do so.

Reply:
Yeah, as long as we have a list of folders to retrieve from the provided folder, then they have the tools they need. We can always open an issue to track it as a future improvement. I just know that I've had to write that code before, and it's not fun and prone to issues, so if we can just provide a handy built-in way, then it could be useful, but yeah not super important at the moment.

Reply:
I'd rather track an issue to handle the recursion than have this PR be dependent on that code block. It will need a fair bit of testing, so let's leave it out for now so I can get moving on the Graph related helpers first :)